src/values/WordCombination.js

```diff
@@ -1,180 +1,69 @@
-import { forEach } from "lodash-es";
-import { has } from "lodash-es";
-
-/**
- * Returns whether or not the given word is a function word.
- *
- * @param {string} word The word to check.
- * @param {Function} functionWords The function containing the lists of function words.
- * @returns {boolean} Whether or not the word is a function word.
- */
-function isFunctionWord( word, functionWords ) {
-	return -1 !== functionWords.indexOf( word.toLocaleLowerCase() );
-}
-
 /**
  * Represents a word combination in the context of relevant words.
  *
  * @constructor
  *
- * @param {string[]} words The list of words that this combination consists of.
+ * @param {string} word The word.
+ * @param {string} stem The stem / base form of the word.
  * @param {number} [occurrences] The number of occurrences, defaults to 0.
- * @param {Function} functionWords The function containing the lists of function words.
  */
-function WordCombination( words, occurrences, functionWords ) {
-	this._words = words;
-	this._length = words.length;
+function WordCombination( word, stem, occurrences ) {
+	this._word = word;
+
+	if ( stem ) {
+		this._stem = stem;
+	} else {
+		this._stem = word;
+	}
+
 	this._occurrences = occurrences || 0;
-	this._functionWords = functionWords;
 }
 
-WordCombination.lengthBonus = {
-	2: 3,
-	3: 7,
-	4: 12,
-	5: 18,
-};
-
 /**
- * Returns the base relevance based on the length of this combination.
- *
- * @returns {number} The base relevance based on the length.
+ * Sets the word to the word combination.
+ * @param {string} word The word to set.
```

Comment:
Thought not strictly needed, I would vote for separation here.

```suggestion
 *
 * @param {string} word The word to set.
 *
```

Reply:
👍